experimental/dds/tree2/src/core/undo/undoRedoManager.ts

```diff
@@ -235,6 +240,35 @@ export class UndoRedoManager<TChange, TEditor extends ChangeFamilyEditor> {
 			return;
 		}
 
+		// If the rebase did not produce any type of change to the tree, the new commits
+		// can be replaced directly without updating repair data
+		if (change === undefined) {
+			const undoables = new Map<RevisionTag, GraphCommit<TChange>>();
+			const redoables = new Map<RevisionTag, GraphCommit<TChange>>();
+
+			// Distinguish which reversible stack each new commit is in
+			for (const commit of newCommits) {
+				const type = originalUndoRedoManager.commitTypes.get(commit.revision);
+				if (type !== undefined) {
```

Comment:
This might be incorrect, but I think that for a rebase we expect `type` to always be defined, and for a merge we expect `type` to never be defined. If that is true, can this be refactored a little better? The merge case doesn't need to do very much.

Reply:
The types are shared between all URM clones so type should be defined for any commit that any branch has seen. Which is what we want in order to understand how to treat commits on branches they haven't yet seen them.

Reply:
Oh yea that's right, I forgot that the map is shared. In that case, shouldn't `type` always be defined? In what cases is there a commit on a branch that doesn't have a commit type recorded?

Reply:
Oh actually, remote edits will not have a type. Is it possible that they would be included in the new commits, though?

Reply:
Great point. Let's see... that would happen if you:

1. Merge the trunk into another branch
2. Rebase another branch over the trunk

(1) Does not happen. (2) does happen, because we rebase the local branch over the trunk. We only ever rebase over one commit, but if that commit is from a remote client then `type` will be undefined. Good catch!